### PR TITLE
[GEOMETRY] fixes UBSAN build errors

### PR DESCRIPTION
**Pull request by smuzaffar**
Fixes for UBSAN IBs build errors

https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/slc7_amd64_gcc820/www/fri/11.2.UBSAN-fri-23/CMSSW_11_2_UBSAN_X_2020-05-22-2300

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by ianna** [comment]
+1

File: Geometry/HcalTowerAlgo/test/HcalCellParameterDump.cc [hunk 1]
```diff
@@ -21,13 +21,13 @@ class HcalCellParameterDump : public edm::one::EDAnalyzer<> {
...
 private:
-  static const int detMax_ = 4;
+  static constexpr int detMax_ = 4;
   int subdet_;
 };
 
 HcalCellParameterDump::HcalCellParameterDump(const edm::ParameterSet& iConfig) {
-  subdet_ = std::min(detMax_, std::max(iConfig.getParameter<int>("SubDetector"), 1));
-  subdet_ = std::min(detMax_, std::max(subdet_, 1));
+  subdet_ = std::min(HcalCellParameterDump::detMax_, std::max(iConfig.getParameter<int>("SubDetector"), 1));
```
Comment (cvuosalo) [inline Geometry/HcalTowerAlgo/test/HcalCellParameterDump.cc:29]: @smuzaffar Why is the `HcalCellParameterDump::` qualification necessary? There is no other `detMax_` that could cause confusion, and the constructor certainly has access to its own data members.

Comment (smuzaffar) [reply Geometry/HcalTowerAlgo/test/HcalCellParameterDump.cc]: you are right @cvuosalo , I started with adding the namespace but then it turned out the `constexpr` should fix the issue. I have updated the PR

**Comment by smuzaffar** [comment]
please test

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**